### PR TITLE
Change directory check in favor of class check to validate if Sensei is active

### LINKED ISSUE #56: Change how Sensei dependencies are checked
Originally reported here: 1376873-zen

The Sensei Course Progress widget was not showing under Appearance > Widgets , so couldn't be used. We tested for conflicts, deactivated everything except Sensei and the Course Progress widget as well as with and without WooCommerce installed in case was something to do with required dependencies https://github.com/woocommerce/sensei-course-progress/blob/master/sensei-course-progress.php for this to work, but we still had no luck and we were unable to replicate a my test site.

After further investigation by @jom  seems that dependencies are checked by validating specific directories, which made the extension to fail to load in this case as the Sensei plugin was installed in `plugins/woo-themes-sensei` instead of `plugins/woothemes-sensei`

This can be seen here: https://github.com/woocommerce/sensei-course-progress/blob/2ac98ab2c5a4118907b65cd8c6cfff40601d5549/woo-includes/class-woothemes-sensei-dependencies.php#L23

Maybe a better way to check this would be via class `class_exists()` instead ?

As a side note, after a first glance to other Sensei extensions code seems that the same dependency check happens as well but I didn't want to start to open issues on other repos until a direction is taken first here.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Work in progress: Fixes #56 

Changes how to check if Sensei is active by removing the specific directory check in favor of a  classes check instead.

To do:
- [x] Changes and testing for `is_multisite()`
- [x] Add `admin_notices` when Sensei is not active

Any advice is welcome, first PR here :)

### CODE REVIEW

**Comment by iamgabrielma** [comment]
While relying in `class_exists()` would be the best solution for checking dependencies, this seems to not work for multisites (or I'm not sure how to make it work), so the initial file directory check cannot be entirely removed. Re-adding a `class_exists()` check additionally to the statement, and leaving the directory check covers all cases (both single sites and multisites)

**Comment by donnapep** [comment]
I tested WPJM and its extensions on a single and multi-site install. They exhibited the same behaviour in both environments when deactivating WPJM. They displayed a notice similar to the following, but remained activated:

screen shot 2018-09-20 at 7 34 29 am

I also tried with the WC Stripe extension and it did a similar thing. So it seems safe to mimic what the [WPJM extensions are doing](https://github.com/Automattic/wp-job-manager-alerts/blob/0ab8f89da983923d05ea5c68d4fc8872c6b03d05/wp-job-manager-alerts.php#L123) or what [WC Stripe](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/d329fc2bee2e02be050abb3da318b16e4297864d/woocommerce-gateway-stripe.php#L38) is doing.

**Comment by iamgabrielma** [comment]
I was having some problems with the class check, which was resolved by wrapping the Sensei class check with the `plugins_loaded` hook. Seems that the check was being performed before Sensei was loaded, and this was returning false negatives.

I've also removed `woo-includes/class-woothemes-sensei-dependencies.php` as now does not performs any validation.

Tested in both single sites and multisite and everything seems working fine.

File: sensei-course-progress.php [hunk 1]
```diff
@@ -29,34 +59,14 @@
...
-	 * @since  1.0.0
-	 * @return object Sensei_Course_Progress
-	 */
-	function Sensei_Course_Progress() {
-		return Sensei_Course_Progress::instance( __FILE__, '1.0.7' );
-	}
+function display_activation_error( $message ) {
+	
+	echo '<div class="error">';
+	echo '<em>Sensei Course Progress</em> requires Sensei to be installed and activated.';
```
Comment (donnapep) [inline sensei-course-progress.php]: This text needs to be translated.

File: sensei-course-progress.php [hunk 2]
```diff
@@ -16,6 +16,36 @@
 
 if ( ! defined( 'ABSPATH' ) ) exit;
 
+/**
+ * Sensei Detection
+ */
+function sensei_course_progress_init(){
```
Comment (donnapep) [inline sensei-course-progress.php]: This function needs to have the whitespace fixed up as per the [PHP coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage).

**Comment by donnapep** [comment]
Closing in favour of #68.

**Pull request closed without merging**